android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/VisualNavigationViewConfig.kt

```diff
@@ -0,0 +1,65 @@
+package com.stadiamaps.ferrostar.maplibreui.extensions
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.unit.LayoutDirection
+import com.mapbox.mapboxsdk.geometry.LatLngBounds
+import com.maplibre.compose.camera.CameraState
+import com.maplibre.compose.camera.MapViewCamera
+import com.maplibre.compose.camera.models.CameraPadding
+import com.stadiamaps.ferrostar.composeui.config.CameraControlState
+import com.stadiamaps.ferrostar.composeui.config.VisualNavigationViewConfig
+import com.stadiamaps.ferrostar.core.NavigationUiState
+import com.stadiamaps.ferrostar.core.boundingBox
+import com.stadiamaps.ferrostar.maplibreui.NavigationViewMetrics
+
+@Composable
+fun VisualNavigationViewConfig.cameraControlState(
```

Comment:
1. This is as good a place as any to drop this I think... I think an extension is better than a free function, but there's no place we colud put this that would remove a substantial number of parameters.
2. Note that this can't live in the same file as the declaration in the `composeui` module, since almost everything about it is MapLibre specific.